app/mattermost.js

```diff
@@ -1,660 +1,473 @@
-            });
-        }
-    };
-
-    // We need to wait for hydration to occur before load the router.
-    listenForHydration = () => {
-        const {dispatch, getState} = this.store;
-        const state = getState();
-        if (!this.isConfigured) {
-            this.configurePushNotifications();
-        }
-
-        if (state.views.root.hydrationComplete) {
-            const orientation = Orientation.getInitialOrientation();
-            const {credentials, config} = state.entities.general;
-            const {currentUserId} = state.entities.users;
-            const isNotActive = AppState.currentState !== 'active';
-            const notification = PushNotifications.getNotification();
-
-            this.unsubscribeFromStore();
-
-            if (this.deviceToken) {
-                // If the deviceToken is set we need to dispatch it to the redux store
-                setDeviceToken(this.deviceToken)(dispatch, getState);
-            }
+const handleOrientationChange = (orientation) => {
+    const {dispatch} = store;
+    if (DeviceInfo.isTablet()) {
+        dispatch(setDeviceAsTablet());
+    }
 
-            if (orientation) {
-                this.orientationDidChange(orientation);
-            }
+    dispatch(setDeviceOrientation(orientation));
+    setTimeout(() => {
+        dispatch(calculateDeviceDimensions());
+    }, 100);
+};
+
+export const handleManagedConfig = async (serverConfig) => {
+    const {dispatch, getState} = store;
+    const state = getState();
+
+    let authNeeded = false;
+    let blurApplicationScreen = false;
+    let jailbreakProtection = false;
+    let vendor = null;
+    let serverUrl = null;
+    let username = null;
+
+    if (LocalConfig.AutoSelectServerUrl) {
+        handleServerUrlChanged(LocalConfig.DefaultServerUrl)(dispatch, getState);
+        app.setAllowOtherServers(false);
+    }
 
-            if (config) {
-                this.configureAnalytics(config);
+    try {
+        const config = await avoidNativeBridge(
+            () => {
+                return true;
+            },
+            () => {
+                return StartTime.managedConfig;
+            },
+            () => {
+                return mattermostManaged.getConfig();
             }
-
-            if (credentials.token && credentials.url) {
-                Client4.setToken(credentials.token);
-                Client4.setUrl(stripTrailingSlashes(credentials.url));
+        );
+        if (config && Object.keys(config).length) {
+            app.setEMMEnabled(true);
+            authNeeded = config.inAppPinCode && config.inAppPinCode === 'true';
+            blurApplicationScreen = config.blurApplicationScreen && config.blurApplicationScreen === 'true';
+            jailbreakProtection = config.jailbreakProtection && config.jailbreakProtection === 'true';
+            vendor = config.vendor || 'Mattermost';
+
+            if (!state.entities.general.credentials.token) {
```

Comment:
what happens if at this point re-hydration hasn't completed yet? aren't you storing this in a keychain?

Reply:
I am storing in Keychain/KeyStore, but this is fine.  Since `handleManagedConfig` is called once rehydration is complete.